connect-inject/handler.go

```diff
@@ -393,6 +478,76 @@ func (h *Handler) defaultAnnotations(pod *corev1.Pod, patches *[]jsonpatch.JsonP
 	return nil
 }
 
+func (h *Handler) consulNamespace(ns string) string {
+	// Return an empty string if namespaces aren't enabled. This
+	// functionality is used for init and sidecar container creation.
+	if !h.EnableNamespaces {
+		return ""
+	}
+
+	// Mirroring takes precedence
+	if h.EnableNSMirroring {
+		return fmt.Sprintf("%s%s", h.MirroringPrefix, ns)
+	} else {
+		return h.ConsulNSName
+	}
+}
+
+func (h *Handler) checkAndCreateNamespace(ns string) error {
+	// Check if the Consul namespace exists
+	namespaceInfo, _, err := h.ConsulClient.Namespaces().Read(ns, nil)
+	if err != nil {
+		return err
+	}
+
+	// If not, create it
+	if namespaceInfo == nil {
+		consulNamespace := api.Namespace{
+			Name:        ns,
+			Description: "Auto-generated by a Connect Injector",
+			// TODO: when metadata is added to the api
+			// Meta:        map[string]string{"external-source": "kubernetes"},
+		}
+
+		_, _, err = h.ConsulClient.Namespaces().Create(&consulNamespace, nil)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (h *Handler) checkAndCreateBindingRule(ns string) error {
+	// Check if the binding rule exists in this namespace
+	existingRules, _, err := h.ConsulClient.ACL().BindingRuleList(h.AuthMethod, &api.QueryOptions{Namespace: ns})
+	if err != nil {
+		return err
+	}
+
+	// When there are no binding rules for the current namespace, one needs to be created
+	if len(existingRules) == 0 {
+		// Note: This description uses the AuthMethod name which differs from binding rules
+		// create in the ACL bootstrap method where the release name is used. It didn't seem
```

Comment:
This was changed due to the fullname support PR. It's now just `Consul default binding rule` I think

Reply:
I agree it's not super important due to the name.